### DISCUSSION TITLE
Repeatable abilities

### DISCUSSION THREAD

**Discussion by pmdoll** [mitre/caldera#2403] [Q&A]
Hello,
I am interested in executing abilities multiple times in my planner. For Ex: I want to execute "Check Python" ability multiple times. 

"How to build Planners?" talks about 

```
NOTE: For planners intended to be used with profiles containing repeatable abilities, 
allow_repeatable_abilities: True
must be added to the planner YAML file. Otherwise, CALDERA will 
default the value to False and assume the planner does not support repeatable abilities.
```

Batch planner has this field set to True but for any given "operation", this planner executes any ability only once. Then trying to run "Potential Link" has the button greyed out with a message "Link is unavailable or has been already added"

Is this a bug or am I incorrect in my understanding of `allow_repeatable_abilities: True`?

Thanks

**Comment by ArtificialErmine** [comment]
Hey @pmdoll - thanks for reaching out! I think what might be going here is a slight bit of confusion regarding repeatability, from an ability and planner standpoint. Repeatable abilities are specially marked abilities (made by adding a `repeatable: True` key to an ability), and these are the only abilities that will be repeated. Generic abilities aren't repeated endlessly as a safety mechanism to keep things from spinning in a loop forever - repeatable abilities and planners (used together) override that safety. This also explains why the "Potential Link" option is disabled - the ability in question isn't repeatable, regardless of whether or not the planner is configured to support repeatable actions.

You can find more information about this keyword for abilities [here](https://github.com/mitre/fieldmanual/blob/master/sphinx-docs/Basic-Usage.md) (these haven't been populated to the live instance of readthedocs just yet due to the impending 4.0 full release).

Hopefully that helps some. Let me know if you have any further questions, and I'll try to elaborate as I can.
